tests/validators/test_model.py

```diff
@@ -1164,3 +1164,166 @@ def f(values_or_values_and_fields_set: Any, *args: Any) -> Any:
     assert m.b == 2
     assert m.__pydantic_fields_set__ == {'a', 'b'}
     assert calls == [call1, call2]
+
+
+def test_custom_init():
+    calls = []
+
+    class Model:
+        __slots__ = '__dict__', '__pydantic_fields_set__'
```

Comment:
```suggestion
        __slots__ = '__dict__', '__pydantic_extra__', '__pydantic_fields_set__'
```

and probably worth adding a test for `__pydantic_extra__`.